src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -907,6 +920,55 @@ try
 }
 CATCH_LOG_RETURN_FALSE()
 
+#pragma warning(push)
+#pragma warning(disable : 26477) // Suppress USE_NULLPTR_NOT_CONSTANT
+
+// Routine Description:
+// - Given a file URI string, attempts to parse the URI encoded.
+//   The spec of the file URI is https://freedesktop.org/wiki/Specifications/file-uri-spec/.
+//   The file URIs are of the form:
+//        file://<hostname>/<path>
+bool OutputStateMachineEngine::_ParseFileUri(const std::wstring_view string,
+                                             std::wstring& hostname,
+                                             std::wstring& path) const
+{
+    if (string.size() < 8)
+    {
+        return false;
+    }
+
+    const auto prefix = string.substr(0, 7);
+    if (prefix.compare(L"file://") != 0)
+    {
+        // We don't support URI format other than "file://".
+        return false;
+    }
+
+    hostname.clear();
+    path.clear();
+
+    size_t current = 7;
+    const size_t nextSlash = string.find(L"/", current);
+    if (nextSlash == std::wstring::npos)
+    {
+        // Invalid URI. Ignore it.
+        return false;
+    }
+
+    hostname = string.substr(current, nextSlash - current);
+    current = nextSlash;
+    std::wstring _path = std::wstring(string.substr(current, std::wstring::npos));
+    if (SUCCEEDED(UrlUnescapeInPlace(_path.data(), URL_ESCAPE_AS_UTF8 | URL_DONT_UNESCAPE_EXTRA_INFO)))

```

Comment:
The `URL_UNESCAPE_AS_UTF8` is only available on Win8+, where as `URL_ESCAPE_AS_UTF8` is available on Win7+. And we have:

```cpp
#define URL_UNESCAPE_AS_UTF8            URL_ESCAPE_AS_UTF8
```

But I'm not sure the flag will do anything on Win7, though.

Reply:
I think the flag `URL_UNESCAPE_AS_UTF8` is very important since URLs escaped by Linux *will* require this to be correctly decoded. See test cases below.